### PR TITLE
Update client method signatures

### PR DESCRIPTION
**Pull request by blowmage**
* Allow request objects to be passed as well as named arguments.
* Individual request field values must now used named arguments.

[refs #69]

### CODE REVIEW

File: gapic-generator-cloud/templates/cloud/client/method/def/_lro.erb [hunk 1]
```diff
@@ -2,16 +2,32 @@
...
 <%- end -%>
+#   @param options [Google::Gax::CallOptions]
+#     Overrides the default settings for this call, e.g, timeout, retries, etc.
 #
-<%- method.fields.each do |field| -%>
-# @param <%= field.name %> [<%= field.doc_types %>]
-<%- if field.doc_description -%>
-<%= indent field.doc_description, "#   " %>
+<%- arg_list = method.arguments.map { |arg| "#{arg.name}: nil"}.join ", " -%>
+# @overload client_method(<%= arg_list %>, options: nil)
```
Comment (landrito) [inline gapic-generator-cloud/templates/cloud/client/method/def/_lro.erb]: This needs to be wrapped in erb brackets.

File: gapic-generator-cloud/templates/cloud/client/method/def/_normal.erb [hunk 2]
```diff
@@ -2,16 +2,32 @@
...
 <%- end -%>
+#   @param options [Google::Gax::CallOptions]
+#     Overrides the default settings for this call, e.g, timeout, retries, etc.
 #
-<%- method.fields.each do |field| -%>
-# @param <%= field.name %> [<%= field.doc_types %>]
-<%- if field.doc_description -%>
-<%= indent field.doc_description, "#   " %>
+<%- arg_list = method.arguments.map { |arg| "#{arg.name}: nil"}.join ", " -%>
+# @overload client_method(<%= arg_list %>, options: nil)
```
Comment (landrito) [inline gapic-generator-cloud/templates/cloud/client/method/def/_normal.erb]: Similarly client_method should be wrapped in erb tags.

File: gapic-generator-cloud/templates/cloud/client/method/def/_lro.erb (hunk 1 above)
Comment (landrito) [reply gapic-generator-cloud/templates/cloud/client/method/def/_lro.erb]: Unless I am misunderstanding `@overload`, client_method should be <%= method.name %>.

Comment (blowmage) [reply gapic-generator-cloud/templates/cloud/client/method/def/_lro.erb]: LOL! Yeah, that is a mistake. That is something from my cheat sheet when I was implementing this. It needs to be `<%= method.name %>`. Fixing now.

Comment (blowmage) [reply gapic-generator-cloud/templates/cloud/client/method/def/_lro.erb]: Fixed.

File: gapic-generator-cloud/templates/cloud/client/method/def/_normal.erb (hunk 2 above)
Comment (blowmage) [reply gapic-generator-cloud/templates/cloud/client/method/def/_normal.erb]: Fixed.

**Comment by blowmage** [comment]
The output files don't have any changes, CI is passing. The gem name's last node is what is being used right now and it is correct for all the gems. But once we get the garbage proto added to cloud we will use the `env_prefix` configuration.

**Pull request merged**